### PR TITLE
Turn off auto optimise store on mac

### PR DESCRIPTION
**Pull request by Hoverbear**
##### Description

Addresses https://github.com/DeterminateSystems/nix-installer/issues/449#issuecomment-1551782281

##### Checklist

- [ ] Formatted with `cargo fmt`
- [ ] Built with `nix build`
- [ ] Ran flake checks with `nix flake check`
- [ ] Added or updated relevant tests (leave unchecked if not applicable)
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] Linked to related issues (leave unchecked if not applicable)

##### Validating with `install.determinate.systems`

If a maintainer has added the `upload to s3` label to this PR, it will become available for installation via `install.determinate.systems`:

```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/pr/$PR_NUMBER | sh -s -- install
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ Differing from the current official [Nix](https://github.com/NixOS/nix) installe
 * In `nix.conf`:
   + the `auto-allocate-uids`, `nix-command` and `flakes` features are enabled
   + `bash-prompt-prefix` is set
-  + `auto-optimise-store` is set to `true`
+  + `auto-optimise-store` is set to `true` (On Linux only)
   * `extra-nix-path` is set to `nixpkgs=flake:nixpkgs`
   * `auto-allocate-uids` is set to `true`.
```
Comment (cole-h) [inline README.md:37]: While we're here, I think auto uid is also linux-only? (and above when listing the features we enable)

Comment (Hoverbear) [reply README.md]: Oh gosh you're right!

File: README.md [hunk 2]
```diff
@@ -32,9 +32,9 @@ Differing from the current official [Nix](https://github.com/NixOS/nix) installe
 * In `nix.conf`:
   + the `auto-allocate-uids`, `nix-command` and `flakes` features are enabled
```
Comment (cole-h) [inline README.md:33]: Here too
```suggestion
  + the `nix-command`, `flakes`, and `auto-allocate-uids` (On Linux only) features are enabled
```

**Review by cole-h — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Hoverbear) [reply README.md:33]: We do actually leave the feature enabled on Mac...

Comment (cole-h) [reply README.md:33]: .......ah

File: README.md [hunk 3]
```diff
@@ -32,9 +32,9 @@ Differing from the current official [Nix](https://github.com/NixOS/nix) installe
...
   + `bash-prompt-prefix` is set
-  + `auto-optimise-store` is set to `true`
+  + `auto-optimise-store` is set to `true` (On Linux only)
```
Comment (SuperSandro2000) [inline README.md:35]: This should be turned off by default on linux, too since it slows down substitution significantly on big stores.

Comment (grahamc) [reply README.md:35]: Can you provide more information? How do big stores cause the substitution to slow down? I could imagine it if the store path is large, but not sure about stores.

Comment (SuperSandro2000) [reply README.md:35]: Yeah, typo, when downloading many or big things from the cache or remote builder while having auto-optimise-store turned on, it noticeable taker longer to complete.

**Pull request merged**